src/main/java/dev/nmarulo/depensaapp/security/JwtAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,50 @@
+package dev.nmarulo.depensaapp.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.nmarulo.depensaapp.commons.dtos.ErrorRes;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationEntryPoint;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint {
+    
+    private final AuthenticationEntryPoint delegate = new BearerTokenAuthenticationEntryPoint();
+    
+    private final ObjectMapper mapper;
+    
+    @Override
+    public void commence(HttpServletRequest request,
+                         HttpServletResponse response,
+                         AuthenticationException authException) throws IOException, ServletException {
+        
+        this.delegate.commence(request, response, authException);

```

Comment:
Please can you explain this line to me.

Reply:
Sure thing! 

We are looking for errors related to `bearerTokens` now in this case the only `AuthenticationEntryPoint` which handles this is the `BearerTokenAuthenticationEntryPoint` (as mentioned in the doc). Now, we don't want to skip the error resolution done by the `BearerTokenAuthenticationEntryPoint`, in an ideal world we could have extended `BearerTokenAuthenticationEntryPoint` and customized the `commence` method. However, we can't as the `BearerTokenAuthenticationEntryPoint` is a final class. So, instead of inheritance we opted for composition, we delegate the error resolution to `BearerTokenAuthenticationEntryPoint` first then handle the `response` body ourselves. This also makes our work simpler as we are not having to handle the response header (`WWW-Authenticate`) ourselves, it is already done by the `BearerTokenAuthenticationEntryPoint` class.

Please let me know if I was clear enough or if more explanation is needed?

Reply:
Thank you, I understand.